pkg/kubectl/history.go

```diff
@@ -257,30 +264,42 @@ func (h *StatefulSetHistoryViewer) ViewHistory(namespace, name string, revision
 	})
 }
 
-// controlledHistories returns all ControllerRevisions controlled by the given DaemonSet
-func controlledHistories(extensions clientextensionsv1beta1.ExtensionsV1beta1Interface, apps clientappsv1beta1.AppsV1beta1Interface, namespace, name string) (*extensionsv1beta1.DaemonSet, []*appsv1beta1.ControllerRevision, error) {
+func retrieveVersionedDaemonSet(extensions clientextensionsv1beta1.ExtensionsV1beta1Interface, namespace, name string) (*extensionsv1beta1.DaemonSet, error) {
 	ds, err := extensions.DaemonSets(namespace).Get(name, metav1.GetOptions{})
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to retrieve DaemonSet %s: %v", name, err)
+		return nil, fmt.Errorf("failed to retrieve DaemonSet %s: %v", name, err)
 	}
+	return ds, nil
+}
+
+func retrieveVersionedStatefulSet(apps clientappsv1beta1.AppsV1beta1Interface, namespace, name string) (*appsv1beta1.StatefulSet, error) {
```

Comment:
@foxish I refactored `controlledHistories` and `controlledSSHistories` functions by introducing two new functions: `retrieveVersionedDaemonSet` and `retrieveVersionedStatefulSet`. I am not sure whether it is possible to support polymorphism for `extensions.DaemonSets(ns)...` and `apps.StatefulSets(ns)...`